akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -1070,6 +1081,12 @@ abstract class ShardCoordinator(
         regionTerminationInProgress -= ref
         aliveRegions -= ref
         allocateShardHomesForRememberEntities()
+        if (ref.path.address.hasLocalScope && waitingForLocalRegionToTerminate) {
```

Comment:
Isn't that enough?

```suggestion
        if (waitingForLocalRegionToTerminate) {
```

Reply:
It could be some other shard region terminating, we want to wait until the local one has completed.